iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/command_line_parser.hpp

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/cxx/convert.hpp"
+#include "iceoryx_hoofs/cxx/expected.hpp"
+#include "iceoryx_hoofs/cxx/string.hpp"
+#include "iceoryx_hoofs/cxx/vector.hpp"
+#include "iceoryx_hoofs/platform/platform_settings.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+namespace cxx
```

Comment:
Actually, this is not from any kind of C++ standard or C++ core guideline - it is a little bit Rust inspired but thats all.
And I do not abstract any kind of posix functionality here, it is more a C argument abstraction to simplify the handling of `argc` and `argv` from `int main(int argc, char *argv[])`.

When one implements this by themself one has to parse the command line, has to perform conversion checks overflow checks and all the stuff which ends up in a very long switch case statement with a lot of `if`.

When using this, all of the above is abstracted away.

Reply:
Yes that's correct that it doesn't use getopt directly but this API has still the goal to replace it. We see something similar with the `Allocator` class in posix_wrapper that also doesn't abstract directly a POSIX function.
Since getopt is a POSIX function it would imho make more sense to locate it there,but i don't see that as a blocker for now.